# Computing STR profiles

EHdn performs a genome-wide search for STRs whose size exceeds the read length.
The program does this by collecting the following read-level evidence:
- **Anchored in-repeat reads**: reads that originate inside STRs and their
  mates (called the **anchors**) originate in the surrounding flanking sequence.
- **In-repeat read pairs** are read pairs where both mates originate inside
  the same STR.

The files containing summaries of identified anchored in-repeat reads and
in-repeat read pairs are called **STR profiles**. STR profiles generated by
EHdn are JSONs containing a separate record for each repeat unit (motif).

For example, consider a CGG repeat depicted in the figure below.
![workflow](images/str-profile.png)
There are 7 anchored in-repeat reads and 3 in-repeat read pairs
originating from this repeat. They give rise to the following entry in the STR
profile file.

```json
"CCG": {
    "AnchoredIrrCount": 7,
    "IrrPairCount": 3,
    "RegionsWithIrrAnchors": {
        "chr1:100-1000": 7
    },
    "RepeatUnit": "CCG"
}
```

STR profiles are computed by the `profile` command as follows. 

```bash
ExpansionHunterDenovo profile \
        --reads input.bam \
        --reference reference.fasta \
        --output-prefix output \
        --min-anchor-mapq 50 \
        --max-irr-mapq 40
```


| Parameter         | Description                                     |
|-------------------|-------------------------------------------------|
| --reads           | The BAM/CRAM file to analyze                    |
| --reference       | The FASTA file to which the reads were aligned  |
| --output-prefix   | Common prefix for the output files              |
| --min-anchor-mapq | Minimum MAPQ for anchor reads                   |
| --max-irr-mapq    | Maximum MAPQ for in-repeat reads                |
